app/models/maintenance_tasks/task.rb

```diff
@@ -21,6 +21,9 @@ class NotFoundError < NameError; end
     # @api private
     class_attribute :collection_builder_strategy, default: NullCollectionBuilder.new
 
+    # As defined by JobIteration::EnumeratorBuilder#build_active_record_enumerator_on_records as `columns`
+    class_attribute :cursor_columns, default: nil
```

Comment:
I searched for a way to let subclasses of `MaintenanceTasks::Task` customize how the `enumerator_builder` instantiates the underlying relation, without success. 

Using a `class_attribute` as an alternative. Is it appropriate?

Reply:
This would be overridden by specific tasks, so I think the best thing to do is provide a simple method like `collection` that can be overridden.